### DISCUSSION TITLE
Zigbee2MQTT & aqara-devices pairing with Sonoff ZBBridge 3.0 as router not possible any more (vs. IKEA Tradfri repeater)

### DISCUSSION THREAD

**Discussion by markus-lorenz** [arendst/Tasmota#17873] [Q&A]
Dear Community,
I opened a post on home assistant forum and kindly repost it here for review, if it is interesting to someone.

In short I am not possible anymore to pair a aqara-end-device to a sonoff zbbridge 3.0 as router. Same device acting as the coordinator and the pairing works without any issue. AND I can pair the same aqara-end-device through a IKEA tradfri-repeater.
But no more through a sonoff zbbridge 3.0.

Maybe someone can help to debug and give a hint.

See:
https://community.home-assistant.io/t/zigbee2mqtt-aqara-devices-pairing-with-sonoff-zbbridge-3-0-as-router-not-possible-any-more-vs-ikea-tradfri-repeater/529914

Kind regards,
Markus.

**Comment by Jason2866** [comment]
All relevant Infos to help and investigate are missing. Which Tasmota firmware used? Which device (type, Firmware) tried to pair? Complete logs which shows what happened.

**Reply by sfromis** [reply]
Sonoff does not sell a product called "Sonoff ZBBridge 3.0". Do you mean the "Sonoff ZigBee 3.0" USB dongle, not relevant to Tasmota?

**Reply by markus-lorenz** [reply]
You are right, on the sonoff-website it is just called "zbbridge" without the "3.0":
https://sonoff.tech/product/gateway-and-sensors/zbbridge/ 

So I do NOT talk about the USB-dongle, but the network bridges.

The reseller mediarath describes the product in the first line as 
"SONOFF ZBBRIDGE ZIGBEE 3.0" and I now think you read it like "SONOFF ZBBRIDGE" with "ZIGBEE 3.0".

Sorry and thanks for the input!

**Reply by markus-lorenz** [reply]
@Jason2866 
Thank you for your comment - you are right.
I tried to find all information I could and edited my post.

As I am quite new to tasmota and zigbee(2mqtt) I have hard times finding information about debugging.
I found [https://tasmota.github.io/docs/Commands/#management](https://tasmota.github.io/docs/Commands/#management) and there the mqttlog and weblog.

Going to console of my sonoff-router and -coordinator via https://MySonoffIP/cs 
I then typed
```
mqttlog 4
weblog 4
```

Tried than to pair a device first with my router (failed) and second with the coordinator (success).

The console-output of my router:
```
11:56:13.081 CMD: MqttLog 4
11:56:13.085 RSL: RESULT = {"MqttLog":4}
11:58:30.311 CMD: MqttLog
11:58:30.316 RSL: RESULT = {"MqttLog":4}
12:01:49.882 CMD: weblog
12:01:49.887 RSL: RESULT = {"WebLog":2}
12:01:56.332 CMD: weblog 4
12:01:56.333 SRC: WebConsole from 10.20.0.102
12:01:56.335 CMD: Grp 0, Cmd 'WEBLOG', Idx 1, Len 1, Pld 4, Data '4'
12:01:56.337 RSL: RESULT = {"WebLog":4}
12:01:57.394 CFG: Saved to flash at FB, Count 59, Bytes 4096
12:02:12.507 WIF: Checking connection...
12:02:25.678 WIF: Sending Gratuitous ARP
12:02:32.532 WIF: Checking connection...
12:02:52.514 WIF: Checking connection...
12:02:55.684 CMD: syslog 4
12:02:55.685 SRC: WebConsole from 10.20.0.102
12:02:55.687 CMD: Grp 0, Cmd 'SYSLOG', Idx 1, Len 1, Pld 4, Data '4'
12:02:55.689 RSL: RESULT = {"SysLog":{"4":{"Active":"4"}}}
12:02:56.409 CFG: Saved to flash at FA, Count 60, Bytes 4096
12:03:12.522 WIF: Checking connection...
12:03:25.636 WIF: Sending Gratuitous ARP
12:03:32.538 WIF: Checking connection...
12:03:52.788 WIF: Checking connection...
12:04:12.751 WIF: Checking connection...
12:04:25.656 WIF: Sending Gratuitous ARP
12:04:32.751 WIF: Checking connection...
12:04:52.792 WIF: Checking connection...
12:05:12.758 WIF: Checking connection...
12:05:25.675 WIF: Sending Gratuitous ARP
12:05:32.758 WIF: Checking connection...
```


The console-output of my coordinator:
```
12:06:33.037 CMD: mqttlog
12:06:33.038 SRC: WebConsole from 10.20.0.102
12:06:33.039 CMD: Grp 0, Cmd 'MQTTLOG', Idx 1, Len 0, Pld -99, Data ''
12:06:33.042 RSL: RESULT = {"MqttLog":0}
12:06:37.621 CMD: weblog
12:06:37.622 SRC: WebConsole from 10.20.0.102
12:06:37.624 CMD: Grp 0, Cmd 'WEBLOG', Idx 1, Len 0, Pld -99, Data ''
12:06:37.626 RSL: RESULT = {"WebLog":3}
12:06:46.636 CMD: mqttlog 4
12:06:46.638 SRC: WebConsole from 10.20.0.102
12:06:46.639 CMD: Grp 0, Cmd 'MQTTLOG', Idx 1, Len 1, Pld 4, Data '4'
12:06:46.642 RSL: RESULT = {"MqttLog":4}
12:06:47.467 CFG: Saved to flash at FB, Count 37, Bytes 4096
12:06:50.494 TCP: to MCU/1: 164321A96C2AF57D337894B8D7EA546FB6B14E2485EBD74489F7C66351217E
12:06:50.499 TCP: from MCU: 6243A1A96C2A151938A77E
12:06:50.502 TCP: from MCU: 7243B1A97D312A1652F8B54AD7555593B463E5D8ABEDCE9874FB
12:06:50.503 TCP: from MCU: DF408BF67D5E3F41D17E
12:06:50.506 TCP: to MCU/1: 87009F7E
12:06:50.550 TCP: to MCU/1: 8070787E
12:06:50.749 TCP: to MCU/1: 204021A9762A158FDD7E
12:06:50.752 TCP: from MCU: 0340A1A9762A151B79
12:06:50.754 TCP: from MCU: 7E
12:06:50.803 TCP: to MCU/1: 8160597E
12:06:51.308 CMD: weblog 4
12:06:51.309 SRC: WebConsole from 10.20.0.102
12:06:51.311 CMD: Grp 0, Cmd 'WEBLOG', Idx 1, Len 1, Pld 4, Data '4'
12:06:51.313 RSL: RESULT = {"WebLog":4}
12:06:51.498 CFG: Saved to flash at FA, Count 38, Bytes 4096
12:06:51.502 TCP: from MCU: 7D3340B1A96B2A164FA674EB04AAA76D499DB3D800C3CE67B60B7E
12:06:51.551 TCP: to MCU/1: 82503A7E
12:06:53.604 WIF: Checking connection...
12:06:55.777 TCP: to MCU/1: 324121A9512ACBB47E
12:06:55.780 TCP: from MCU: 2441A1A9512A81E97E
12:06:55.831 TCP: to MCU/1: 83401B7E
12:07:03.047 TCP: to MCU/1: 434621A9FB2AEA4DA66BB5DA55AAC820FB0C42CEACA20BE29CA800ECCC4741A67E
12:07:03.053 TCP: from MCU: 3546A1A9FB2A15173B7E
12:07:03.103 TCP: to MCU/1: 8430FC7E544721A9762AEB0CEA7E
12:07:03.107 TCP: from MCU: 4647A1A9762A1577737E
12:07:03.156 TCP: to MCU/1: 8520DD7E654421A9012A15B159AABC7E
12:07:03.160 TCP: from MCU: 5744A1A9012A156B7D317E
12:07:03.211 TCP: to MCU/1: 8610BE7E764521A92A2ACF7D31146AB5A1A7519377147E
12:07:03.215 TCP: from MCU: 6045A1A92A2AB0FD
12:07:03.216 TCP: from MCU: 7E
12:07:03.265 TCP: to MCU/1: 87009F7E074A21A9602A15B259904B04AAA760099D4E2785C2C87D5EAFFFED9D89FF1E7E
12:07:03.272 TCP: from MCU: 714AA1A9602A151E
12:07:03.274 TCP: from MCU: D97F7E014AB1A97D312A15B658B54AD7581593499C
12:07:03.276 TCP: from MCU: E2D8ABEDCE9874FBDF478BD7803FAB7D317E7D314AB1A9
12:07:03.278 TCP: from MCU: 6B2A15B259904B04AAA760099D4E2707C2CE
12:07:03.281 TCP: to MCU/1: 8070787E
12:07:03.282 TCP: from MCU: 672D867E
12:07:03.323 TCP: to MCU/1: 8160597E82503A7E
12:07:04.985 TCP: from MCU: 214AB1A90D2A45E7A044072DAA
12:07:04.987 TCP: from MCU: D88749ECF626ABED88A77E
12:07:05.035 TCP: from MCU: 314AB1A97D312A15B6
12:07:05.037 TCP: to MCU/1: 83401B7E
12:07:05.038 TCP: from MCU: 58944A24AB5593499CE04B1CBD9B9874DEDA3C98E3743E58A9D7DF4E32F4C4F3C8D64824552688FA
12:07:05.040 TCP: from MCU: 763D81EB758241984607787D33E40B727E
12:07:05.085 TCP: to MCU/1: 8430FC7E
12:07:05.785 TCP: to MCU/1: 144B21A9512A35F37E
12:07:05.788 TCP: from MCU: 424BA1A9512A15BE7E
12:07:05.839 TCP: to MCU/1: 8520DD7E
```

I would be very pleased if you can point me to the right direction of getting correct logs as I think this here is not, what you / we are looking for,...

**Reply by sfromis** [reply]
When it comes to using router firmware, nothing more will be visible in Tasmota, as the router firmware flashed to the coordinator chip is running completely autonomously without any interaction with Tasmota.

**Reply by markus-lorenz** [reply]
Very interesting background, @sfromis.
If I understand it right, a tasmota-flashed device acting as a router will get more or less "non-visible" in his behaviour?
That would raise the conceptual question to me, what is the actual benefit of a router to a repeater within a zigbee-network.
My idea was, that a repeater just receives packages and sends them to the next station: "non-visible" in acting, debugging and logging.
If I now here, a router behaves exactly the same way: for what reason(s) should I use a router and not just 1 coordinator and a bunch of repeater?

**Reply by sfromis** [reply]
Well, calling something a Zigbee "repeater" is nothing more than another phrase for what in standard Zigbee terminology is called a "router", with the purpose being to strengthen the mesh in case of otherwise having sparcity of stable nodes to forward the traffic of devices not in good range of the one coordinator (or limitations in how many can be connected directly). It seems that the main reason for creating a router build for the ZbBridge was for people being able to reuse this box in case of upgrading to a better coordinator device, like one with CC2652 and maybe Ethernet.

I'm not sure of details, but it may be possible to add limited end device functions to a router, basically in the same way as a mains powered relay also acts as a router.
